plugins/Wallet/js/sagas/helpers.js

```diff
@@ -0,0 +1,92 @@
+// Helper functions for the wallet plugin.  Mostly used in sagas.
+
+import BigNumber from 'bignumber.js';
+import { List } from 'immutable';
+import Siad from 'sia.js';
+const uint64max = Math.pow(2, 64);
+
+// siadCall: promisify Siad API calls.  Resolve the promise with `response` if the call was successful,
+// otherwise reject the promise with `err`.
+export const siadCall = (Siad, uri) => new Promise((resolve, reject) => {
+	Siad.call(uri, (err, response) => {
+		if (err) {
+			reject(err);
+		} else { 
+			resolve(response);
+		}
+	})
+})
+
+// Compute the net value and currency type of a transaction.
+const computeSum = (txn) => {
+	var value = new BigNumber(0);
+	var currency;
+	if (txn.inputs) {
+		for (let i = 0; i < txn.inputs.length; i++) {
+			const input = txn.inputs[i];
+			if (input.walletaddress) {
+				value = value.sub(input.value);
+			}
+			// TODO: Imperfect way to go about determining currency of
+			// transaction from 'miner', 'siacoin', or 'siafund'
+			currency = input.fundtype.split(' ')[0];
+		}
+	}
+	if (txn.outputs) {
+		for (let i = 0; i < txn.outputs.length; i++) {
+			const output = txn.outputs[i];
+			if (output.walletaddress) {
+				value = value.add(output.value);
+			}
+			// TODO: Imperfect way to go about determining currency of
+			// transaction from 'miner', 'siacoin', or 'siafund'
+			currency = output.fundtype.split(' ')[0];
+		}
+	}
+	return {
+		value: Siad.hastingsToSiacoins(value),
```

Comment:
I believe so.  This function was largely lifted out of the old wallet plugin, TODOs included.  `currency` isn't currently used in the UI anyway, however.